Model/ResourceModel/NodeType/Product.php

```diff
@@ -89,4 +97,33 @@ public function fetchConfigData()
     {
         return [];
     }
+
+    /**
+     * @param int $storeId
+     * @param array $productIds
+     * @return array
+     */
+    public function fetchTitleData($storeId = Store::DEFAULT_STORE_ID, $productIds = [])
+    {
+        $metadata = $this->metadataPool->getMetadata(ProductInterface::class);
+        $identifierField = $metadata->getIdentifierField();
+
+        $connection = $this->getConnection('read');
+        $select = $connection
+            ->select()
```

Comment:
Use the product collection (`$productCollection`) instead (your code isn't compatible with the Scheduled Changes feature - Magento Commerce). Have a look at `fetchImageData()`.